### ISSUE TITLE
Class Weight for 3D Targets

### ISSUE THREAD

**Issue by KlaymenGC** [fchollet/keras#1624]
Hello,

I'm trying to add weights to the classes during training, the output shape of the last layer (```softmax```) is ```(batch_size, length, nb_class)``` with one-hot-label of the same size, and ```categorical_crossentropy``` loss is used here. 

I've tried to use ```class_weight``` in ```fit``` and send class weight as dictionary: ```{0:w1, 1:w2, ...}```, but got the error: ```Exception: class_weight not supported for 3+ dimensional targets.```

Could someone suggest a way of doing this in Keras? Any suggestions will be greatly appreciated!

**Comment by KlaymenGC** [comment]
#1573

**Comment by fchollet** [comment]
The concept of 'class' is ill-defined is that case (3D labels). You should
use sample_weights instead, computed however you wish to define the concept
of 'class' in your situation.

**Comment by KlaymenGC** [comment]
@fchollet it seems that ```sample_weight``` only takes in 1D or 2D array, in my case, I'll have to pass a ```sample_weight``` of size ```(batch_size, length, nb_class)```, since the weight for the first dimension will always be the same, I pass the weight of size ```(length, nb_class)```, and got the error ```sample_weight should be the same size of X and y```. Do you think it's possible to do it in Keras?

**Comment by GodOfProbability** [comment]
Have you found the solution yet? I am having similar problem.

**Comment by KlaymenGC** [comment]
I haven't followed the development of Keras for quite a while, but if you want to assign weight to classes, you could try weighting the loss function.

**Comment by GodOfProbability** [comment]
I am finally using sample_weight, as all the output of the LSTM belong to the same class (in my case), hence sample_weight suits my purpose.

**Comment by onkarpandit00786** [comment]
@GodOfProbability 
Can you please tell, how did you construct sample_weights for lstm... I am also facing same issue with 3D data..

**Comment by mongoose54** [comment]
I am having the same issue. Any updates? Running latest Keras. It seems that it has not been fixed.

**Comment by Jeffyrao** [comment]
I am having the issue, any updates?
